### ISSUE TITLE
PopupBridge.mWebView / mContext null after process kill

### ISSUE THREAD

**Issue by herbeth1u** [braintree/popup-bridge-android#26]
### General information

* SDK/Library version: 2.0.0
* Android Version and Device: Multiple devices from multiple OEMs (60% Xiaomi Mi A2 Lite, then BlackBerry, Sony and WIKO), all Android versions (6, 8, 9, 10).

### Issue description

Hello again. After some tests following #25 , we've managed to reproduce the crash so we're creating a new issue as mentioned in the previous issue. The stacktrace as a reminder:
```
Caused by java.lang.NullPointerException: Attempt to invoke virtual method 'boolean android.view.View.post(java.lang.Runnable)' on a null object reference
       at com.braintreepayments.popupbridge.PopupBridge.runJavaScriptInWebView(PopupBridge.java:100)
       at com.braintreepayments.popupbridge.PopupBridge.onBrowserSwitchResult(PopupBridge.java:153)
       at com.braintreepayments.browserswitch.BrowserSwitchFragment.onResume(BrowserSwitchFragment.java:73)
       at androidx.fragment.app.Fragment.performResume(Fragment.java:2747)
       at androidx.fragment.app.FragmentStateManager.resume(FragmentStateManager.java:373)
       at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1199)
       at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1356)
       at androidx.fragment.app.FragmentManager.moveFragmentToExpectedState(FragmentManager.java:1434)
       at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1497)
       at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:2625)
       at androidx.fragment.app.FragmentManager.dispatchResume(FragmentManager.java:2589)
       at androidx.fragment.app.FragmentController.dispatchResume(FragmentController.java:269)
       at androidx.fragment.app.FragmentActivity.onResumeFragments(FragmentActivity.java:478)
       at androidx.fragment.app.FragmentActivity.onPostResume(FragmentActivity.java:467)
       at androidx.appcompat.app.AppCompatActivity.onPostResume(AppCompatActivity.java:195)
       at android.app.Activity.performResume(Activity.java:7184)
       at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3694)
       at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3766)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2939)
       at android.app.ActivityThread.-wrap11(ActivityThread.java)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1615)
       at android.os.Handler.dispatchMessage(Handler.java:105)
       at android.os.Looper.loop(Looper.java:231)
       at android.app.ActivityThread.main(ActivityThread.java:6749)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
```

### Steps to reproduce
FYI we are using PopupBridge for Paypal.
1. Initialize the SDK as mentioned in the docs (at this point we've made sure our webview cannot be null) :
```kotlin
webView?.let { PopupBridge.newInstance(activity, it) }
```
2. Click on "Buy with Paypal" : the Paypal Chrome Tabs opens.
3. Log in to Paypal if necessary but don't click on "Buy".
4. Trigger a process kill ([see here](https://medium.com/inloopx/android-process-kill-and-the-big-implications-for-your-app-1ecbed4921cb) for more information on how to simulate a process kill).
5. Go back to the application : the Paypal Chrome Tabs restarts and loads again properly.
6. Click on "Buy".
7. The `PopupBridge` fragment is recreated, but its data initialized in `newInstance` and not saved in a `Bundle` have been erased (including `mWebView` and `mContext`). When going in `onResume`, the NPE occurs.

Again feel free to ask for more details!
Thank you.

**Comment by sshropshire** [comment]
Hi @herbeth1u thanks again for using the sdk. I tried reproducing in our demo application and am unable to reproduce. I do however see the concern. May I trouble you for a screen-capture as well?

Theoretically, I'm unable to imagine how `WebView` can become null if you're certain that a non-null reference is being passed into the `PopupBridge::newInstance` method.

**Comment by herbeth1u** [comment]
Hi @sshropshire , we have recorded a video showcasing the crash. For obvious confidentiality issues, we would like to forward it to you in a private message. However it doesn't seem to be possible to send a PM on Github.
Do you perhaps have a public email address or any other private means of communication so we can send you the video?
Thanks.

**Comment by sshropshire** [comment]
Hello @herbeth1u apologies for the late response. We are able to reproduce similar behavior in our demo application, and it has to do with the deep link back into the PopupBridgeActivity. It seems that Android isn't invoking the deep link after a process kill. We are investigating the issue.

**Comment by sarahkoop** [comment]
This fix is released in 4.0.0-beta2.

**Comment by sshropshire** [comment]
Version 4.0.0 is now generally available.
